infra/README.md

```diff
@@ -0,0 +1,180 @@
+This is a work in progress to use Terraform to manage our cloud
+infrastructure.
+
+Instructions:
+
+- You will need a GCP project.  Configure `gcloud` to point at your project:
+
+   ```
+   gcloud config set project <gcp-project-id>
+   gcloud config set compute/zone <gcp-zone>
+   ```
+
+- Delete the default network if it exists.
+
+- Create a service account for Terraform with Owner role, create a
+  service account key and place it in
+  `$HOME/.hail/terraform_sa_key.json`.
+
+- Enable the the GCP services needed by Hail:
+
+   ```
+   gcloud services enable \
+       compute.googleapis.com \
+       cloudresourcemanager.googleapis.com \
+       servicenetworking.googleapis.com \
+       sqladmin.googleapis.com \
+       container.googleapis.com \
+       serviceusage.googleapis.com \
+       dns.googleapis.com \
+       logging.googleapis.com \
+       cloudprofiler.googleapis.com
+   ```
+
+- Install terraform.
+
+- Create `$HOME/.hail/global.tfvars` that looks like:
+
+   ```
+   gsuite_organization = "<gsuite-organization>"
+   # batch_gcp_regions is a JSON array of string, the names of the gcp
+   # regions to schedule over in Batch.
+   batch_gcp_regions = "<batch-gcp-regions>"
+   gcp_project = "<gcp-project-id>"
+   # gcp_location is the bucket location that spans the regions you're
+   # going to schedule across in Batch.  If you are running on one
```

Comment:
> If you are running on one region, it can just be that region.

Unfortunately that doesn't quite work in the Australian setting. `location = "australia-southeast1"` and `storage_class = "MULTI_REGIONAL"` yields:
> Error: googleapi: Error 400: The combination of locationConstraint and storageClass you provided is not supported for your project, invalid

Unfortunately there's only a US, EU, and ASIA setting for multi-region buckets, so that unfortunately would be pretty expensive from Australia.

Reply:
OK, I renamed gcp_location => batch_logs_bucket_location and added a batch_logs_bucket_storage_class tfvar for this.